src/Microsoft.OData.Edm/Schema/EdmEntityType.cs

```diff
@@ -157,6 +158,37 @@ public void AddKeys(IEnumerable<IEdmStructuralProperty> keyProperties)
                     this.declaredKey = new List<IEdmStructuralProperty>();
                 }
 
+                //The Key property is the last segment in the path segment. 
+                IEdmStructuralPropertyAlias structuralPropertyAlias = property as IEdmStructuralPropertyAlias;
+                if (!string.IsNullOrEmpty(structuralPropertyAlias?.PropertyAlias))

```

Comment:
Can the logic in this block be placed in a location where it can be shared given that it looks similar to that in _src/Microsoft.OData.Edm/Csdl/Semantics/CsdlSemanticsEntityTypeDefinition.cs_?

Reply:
Actually, I think it makes the most sense to go one step further even: the `EdmStructuralPropertyAlias` constructor should do this work, otherwise there are far too many validations that need to happen in order to ensure the `EdmStructuralPropertyAlias` is not malformed.

Reply:
See my comment on the constructor, there just so much validation that would need to happen

Reply:
The `EdmStructuralPropertyAlias` is not the `property` with an alias in this case.

Reply:
@ElizabethOkerio what was the conclusion to this thread?

Reply:
I'm confused a bit. If the property here is already an `IEdmStructuralPropertyAlias` why do you need to possibly duplicate `EdmStructuralPropertyAlias` via `PropertyAliasHelpers.GetKeyProperty()` instead of adding the property directly to the declared key?

Reply:
@habbes The `IEdmStructuralPropertyAlias` is for the first segment but this method  `PropertyAliasHelpers.GetKeyProperty()`  returns `IEdmStructuralPropertyAlias` for the last segment..

Reply:
@habbes  am not sure I understand this. So we are the ones who create the `EdmStructuralPropertyAlias ` once we've gotten to the last segment on the path.

Reply:
Am not sure there is any fast way of getting the last segment.. When we call the `AddKeys` method, we pass in the first segment property and the path. So to get to the last segment we have to traverse the whole path. If we decide on passing the last segment property to the `AddKeys` method, that will mean we are assuming the path is valid and no path traversal is needed.